### ISSUE TITLE
Spark support on remote dataframes

### ISSUE THREAD

**Issue by CJ-Wright** [intake/intake#451]
Are the plans to support `to_spark` for remote dataframes? Currently the `RemoteDataFrame` class's `to_spark` method resolves to the base class where it raises a not implemented error despite the `intake-parquet` and `intake-spark` packages being installed.

@martindurant

**Comment by martindurant** [comment]
That would imply that spark knows how to read from the intake server, so I don't see how.
If you can open locally, you can use spark.

**Comment by CJ-Wright** [comment]
Could we not have the intake server create a spark rdd on the spark cluster and then ship it back to the user?

**Comment by martindurant** [comment]
Yes, you could open something with the spark driver on the server side, but we have no way to "ship" the rdd or make a local sparkable version of it.

**Comment by CJ-Wright** [comment]
Is the returned rdd the actual data or just a reference to the data? If it is a reference can we pickle it and push it over the wire?

**Comment by martindurant** [comment]
It is *not* the data, that would be a dataframe.
If the RDD would make the same sense on the client side, we wouldn't need to pickle or send anything. I think you are really asking for an RDD which executes to download the data from the server, like a delayed function for dask. It might be possible, but someone would have to write it.

**Comment by CJ-Wright** [comment]
Can you elaborate on what you think would need to be written?

**Comment by CJ-Wright** [comment]
As I was reading through the code there seem to be two separate dask (for lack of a better word) clusters when accessing data on a remote server. There is a client cluster that sends a bunch of post requests to the server and the server cluster that reads up the data and then sends it back in chunks.

Would it be possible to have those clusters be the same cluster? That way the intake server itself doesn't need to wait on the data reading, it could return the future itself, which may be computed upon at some later date. Additionally this would make the intake server have a small memory footprint. Currently the intake server has to sit on enough memory to read the chunks in parallel.

Spark handling could work similar, so that a request to the server to read up the data on spark causes the server to send out a spark job. The spark job then returns the rdd which is returned to the user.
